packages/config-array/src/config-array.js

```diff
@@ -599,6 +626,14 @@ export class ConfigArray extends Array {
 		} else {
 			this.push(configs);
 		}
+
+		// select path-handling implementations depending on the base path
+		this.path = getPathImpl(this.basePath);
+
+		// On Windows, `path.relative()` returns an absolute path when given two paths on different drives.
+		// The namespaced base path is useful to make sure that calculated relative paths are always relative.
+		// On Unix, it is identical to the base path.
+		this.namespacedBasePath = this.path.toNamespacedPath(this.basePath);
```

Comment:
Should these field be private?

Reply:
I think yes. We could make them public in the future if there's a need for them.

Reply:
Done in 249540676f3a7486d1f7b4cf8637e142fe99bf7a.